verify_mapping_quality.pl

```diff
@@ -0,0 +1,253 @@
+        "h|help"      => \$help,
+        "m|man"       => \$man
+    );
+    pod2usage(1) if $help;
+    pod2usage(-verbose => 2) if $man;
+
+    unless ( (defined $dir && -e $dir) ) {
+        print "Please specifiy a directory or file name.\n\n";
+        pod2usage(1);
+    }
+    unless ( defined $size ) {
+        print "Please specify the size of the genome.\n\n";
+        pod2usage(1);
+    }
+
+	#set default values if undefined on command line
+    if(!defined $min_depth){
+        $min_depth = $MIN_DEPTH;
+        print "  Using a minumum depth of $MIN_DEPTH\n";
+        print "  Use the --min-depth flag to set custom depth\n\n";
+    }
+    else{
+    	print "Using a min_depth of ".$min_depth."\n";
+    }
+	$log_dir='' if (not defined $log_dir);
+	$min_map=$MIN_MAP if (not defined $min_map);
+	
+	#create the log file to print warnings to
+	open(my $log, '>'.$log_dir.'mapping_percentage.log');
+	
+    my @files;
+    @files = get_bams($dir);
+	
+	#parse results and determine if the pipeline should die, or warnings thrown
+	my @results;
+    @results = verify_percent_coverage( \@files, $size, $min_depth );
+    
+    print $log "==========Reference Mapping Quality===========\n";
+    print $log "NUMBER OF BP's IN REFERENCE GENOME: ".$size."\n";
+    foreach my $result(@results){
+    	my @split = split(',', $result);
+    	my @double = split('%', $split[1]);
+    	print $log "Mapping to reference for isolate ".$split[0]." is ".$split[1]."\n" if $double[0] < $min_map; 
+    }
+   	
+}
+
+#----------------------------------------------------------#
+# Calculate the % Coverage for all .bam files in directory #
+#----------------------------------------------------------#
+sub verify_percent_coverage {
+    my ( $files, $size, $min_depth ) = @_;
+
+    # Get the number of available cores
+    #----------------------------------#
+    my $cores = get_num_cores();
+
+    # Create a Prallel::ForkManager
+    #------------------------------#
+    my $pm = new Parallel::ForkManager($cores);
+
+    my @results;
+    
+    # data structure retrieval and handling
+    $pm -> run_on_finish ( # called BEFORE the first call to start()
+        sub {
+            my ($pid, $exit_code, $ident, $exit_signal, $core_dump, $data_structure_reference) = @_;
+            # retrieve data structure from child
+            if (defined($data_structure_reference)) {  # children are not forced to send anything
+                push @results,${$data_structure_reference};
+            } else {  # problems occuring during storage or retrieval will throw a warning
+                die "Did not recieve data from one of the children\n";
+            }
+        }
+    );
+    
+    
+    # Let's do the heavy lifing
+    #--------------------------#
+    
+    foreach my $file ( @$files ) {
+        $pm->start && next;
+        
+        # Get the actual file name
+        #-------------------------#
+        my $name = fileparse( $file, ".bam" );
```

Comment:
This will not work when using Galaxy. Each file in galaxy has the same name format. ie
dataset34343.dat 
dataset34344.dat 
etc..

When running in galaxy, you will need to get the names from the command line. Example code base will be 
https://irida.corefacility.ca/galaxy-shed/repos/irida/core_pipeline/file/0737c0310cab

The galaxy wrapper (*.xml) file will pass both the sample name and the file path as a single string on the command line
https://irida.corefacility.ca/galaxy-shed/repos/irida/core_pipeline/file/0737c0310cab/vcf2pseudoalignment.xml#l26
so should look like this : --bam Strain1=dataset34343.dat --bam Strain3=dataset34344.dat

GetOpt::long will take care of the rest.

https://metacpan.org/pod/Getopt::Long#Options-with-hash-values

If the script is called by the command line (i.,e passed a directory), you would then do what same as above.